main.cpp

```diff
@@ -0,0 +1,220 @@
+
+        string typeid_name = typeid(value_).name();
+
+        if (typeid_name == typeid(bool).name()) {
+            value.as_bool = static_cast<bool>(value_);
+            type = Type::Bool;
+        }
+        else if (typeid_name == typeid(char).name()) {
+            value.as_char = static_cast<char>(value_);
+            type = Type::Char;
+        }
+        else if (typeid_name == typeid(int).name()) {
+            value.as_int = static_cast<int>(value_);
+            type = Type::Int;
+        }
+        else if (typeid_name == typeid(double).name()) {
+            value.as_double = static_cast<double>(value_);
+            type = Type::Double;
+        }
+        else if (typeid_name == typeid(float).name()) {
+            value.as_float = static_cast<float>(value_);
+            type = Type::Float;
+        }
+        else
+            throw "Bad_cast caught: Fundamental types only!";
+
+        return *this;
+    }
+    ///
+    /// \brief operator =
+    /// \param atObj
+    /// \return
+    ///Copy operator
+    AnyType& operator=(const AnyType atObj){
+
+        if (&atObj == this)
+            return *this;
+        value = atObj.value;
+        type = atObj.type;
+        return *this;
+    }
+    ///
+    /// \brief swap
+    /// \param atObj
+    ///Swaps two objects of AnyType class
+    void swap(AnyType& atObj){
+    if(type == atObj.type){//switch
+        switch(type){
+        case Type::Int:
+            std::swap(this->value.as_int, atObj.value.as_int);
+            break;
+        case Type::Float:
+            std::swap(this->value.as_float, atObj.value.as_float);
+            break;
+        case Type::Double:
+            std::swap(this->value.as_double, atObj.value.as_double);
+            break;
+        case Type::Bool:
+            std::swap(this->value.as_bool, atObj.value.as_bool);
+            break;
+        case Type::Char:
+            std::swap(this->value.as_char, atObj.value.as_char);
+            break;
+        }
+    }
+    std::swap(type, atObj.type);
+}
+    ///
+    /// \brief getType
+    /// \return
+    ///Returns the type of the contained value
+    string getType(){
+
+        switch(type){
+            case Type::Int:
+               return "int";
+            case Type::Float:
+               return "float";
+            case Type::Double:
+               return "double";
+            case Type::Bool:
+               return "bool";
+            case Type::Char:
+               return "char";
+        }
+        return "empty";
+    }
+
+    template <typename T>
+    ///
+    /// \brief getValue
+    /// \return
+    ///Extracts a value should provide type-safe access. If the requested type does not match to stored type exception is thrown
+    T getValue() {
+        char current_type = static_cast<char>(getType()[0]);
+
+        if(current_type == *typeid(T).name())
+            switch(type){
+                case Type::Int:
+                    return static_cast<int>(value.as_int);
```

Comment:
conversion from 'int' to 'T', possible loss of data

Reply:
Resolved